src/main/java/org/testng/reporters/JUnitReportReporter.java

```diff
@@ -81,12 +82,14 @@ public void generateReport(List<XmlSuite> xmlSuites, List<ISuite> suites,
 //      afters.put(es.getKey(), es.getValue().iterator());
 //    }
 
+    SimpleDateFormat df = new SimpleDateFormat("dd MMM yyyy HH:mm:ss");
     for (Map.Entry<Class<?>, Set<ITestResult>> entry : results.entrySet()) {
       Class<?> cls = entry.getKey();
       Properties p1 = new Properties();
       p1.setProperty("name", cls.getName());
       Date timeStamp = Calendar.getInstance().getTime();
-      p1.setProperty(XMLConstants.ATTR_TIMESTAMP, timeStamp.toGMTString());
+      String timestampStr = df.format(timeStamp) + " GMT";
```

Comment:
Ok, but did you check what is the expected or used format from JUnit? 
It looks like `ISO 8601` is expected, so you don't need to imagine a custom pattern.